File: Quivi.Backoffice.Api/Requests/MenuCategories/PatchMenuCategoryRequest.cs
using Quivi.Domain.Entities;
using Quivi.Infrastructure.Apis;

namespace Quivi.Backoffice.Api.Requests.MenuCategories
{
    public class PatchMenuCategoryRequest : ARequest
    {
        public string? Name { get; init; }
        public Optional<string> ImageUrl { get; init; }
        public IDictionary<Language, PatchMenuCategoryLanguage?>? Translations { get; init; }
    }

    public class PatchMenuCategoryLanguage
    {
        public string? Name { get; init; }
    }
}

File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/menuCategories/PatchMenuCategoryRequest.ts
import { Language } from "../Language";

export interface PatchMenuCategoryRequest {
    readonly id: string;
    readonly name?: string;
    readonly imageUrl?: string | null;
    readonly translations?: Record<Language, PatchMenuCategoryTranslation | undefined>;
    readonly menuCategoryIds?: string[];
}

export interface PatchMenuCategoryTranslation {
    readonly name?: string;
}